src/Sarif.UnitTests/TestData/PrereleaseCompatibilityTransformer/Inputs/RunResources.sarif

```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-2.0.0-csd.2.beta.2018-10-10",
+  "version": "2.0.0-csd.2.beta.2018-10-10",
+  "runs": [
+    {
+      "tool": {
+        "name": "CodeScanner",
+        "semanticVersion": "2.1.0"
+      },
+      "resources": {
+        "messageStrings": {"noContent": "This file has no content to review."},
+        "rules": {
+          "SHARED-ID.2": {
+            "id": "SHARED-ID",
+            "configuration": {
+              "defaultLevel": "error",
+              "enabled": false
+            },
+            "messageStrings": { "a": "Review all image content for geopolitically sensitive graphics." }
+          },
+          "SHARED-ID.1": {
+            "id": "SHARED-ID",
+            "messageStrings": { "a": "Review all plaintext content for geopolitically sensitive terms." }
+          }
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SHARED-ID.1",
+          "message": { "messageId": "a" },
+          "locations": [ { "physicalLocation": { "fileLocation": { "uri": "file.txt" } } } ]
+        },
+        {
+          "ruleId": "SHARED-ID.1",
+          "level": "notApplicable",
+          "message": { "messageId": "noContent" },
+          "locations": [ { "physicalLocation": { "fileLocation": { "uri": "empty_file.txt" } } } ]
+        },
+        {
+          "ruleId": "SHARED-ID.2",
+          "message": { "messageId": "a" },
+          "locations": [ { "physicalLocation": { "fileLocation": { "uri": "file.png" } } } ]
+        }
+      ]
+    }
+  ]
+}
```

Comment:
I love how well thought out this example is -- two variants of a rule about "sensitive content", the graphics variant being disabled by default but an error if present, the "empty file" case using a generic message -- really nice.

Reply:
the count of reviewers who would notice this subtlety is drastically smaller than non-members of the set. :)

---
In reply to: [255309492](https://github.com/Microsoft/sarif-sdk/pull/1259#discussion_r255309492) [](ancestors = 255309492)